pkg/utils/pod.go

```diff
@@ -32,7 +32,7 @@ func CheckPodFailed(pod *corev1.Pod) error {
 		return fmt.Errorf("pod %s has failed status", pod.Name)
 	}
 
-	if pod.Status.Phase == corev1.PodPending && isPodReasonUnschedulable(pod) {
+	if pod.Status.Phase == corev1.PodPending && IsPodReasonUnschedulable(pod) {
```

Comment:
anything is fine. Both functions are just on the conditions, even we can just conditions also in these functions.

Reply:
Yeah, anything is fine. My thinking was to keep every un-schedulable condition under a func so that we don't have to call multiple conditions ANDed together at multiple places if required.